Zooming and panning (#6441)
qgis/QGIS-Documentation: docs/user_manual/introduction/general_tools.rst

.. index:: Zoom, Pan, Map navigation
.. _zoom_pan:

Zooming and Panning
-------------------

There are multiple ways to zoom and pan to an area of interest.
You can use either the mouse buttons on the map canvas, the keyboard
on the map canvas, or the menu actions.

> With the Tools in the Toolbar

Author: It felt awkward those lists. I wasn't sure how to present the tools and actions. The table is nice, I'll do it that way.

.............................

Zooming and panning the map can be done with the following tools available in the toolbar:

>   * |pan| :sup:`Pan` will activate the pan tool and allow you to left click anywhere on the map

Reviewer: ```suggestion
  * |pan| :sup:`Pan` allows you to left click anywhere on the map
```
It's somehow expected

    canvas to pan the map at the cursor position. You can also pan the map
    by holding down the left mouse button and dragging the map canvas.

  * |zoomIn| :sup:`Zoom In` and |zoomOut| :sup:`Zoom Out` will activate the Zoom In and Zoom Out tools allowing
    you to left click anywhere on the map canvas to zoom in or zoom out one level. The mouse cursor
    position will be the center of the zoomed area of interest. You can also zoom in
    to an area (or zoom out) by dragging a rectangle on the map canvas with the left mouse button.

>   * |panToSelected| :sup:`Pan Map to Selection` will pan the map to the selected features of all the selected layers in the layer panel.

Reviewer: better use :menuselection:`Layers` panel

  * |zoomToSelected| :sup:`Zoom to Selection` will zoom to the selected features of all the selected layers in the layer panel.

  * |zoomToLayer| :sup:`Zoom to Layer(s)` will zoom to the extent of all the selected layers in the layer panel.

  * |zoomFull| :sup:`Zoom Full` will zoom to the extent of all the visible layers in the map.

  * |zoomLast| :sup:`Zoom Last` will zoom the map to the previous extent in history.

  * |zoomNext| :sup:`Zoom Next` will zoom the map to the next extent in history.

A :guilabel:`Zoom factor` can be set under the
:menuselection:`Settings -->` |options| :menuselection:`Options --> Map tools`

> menu to define the scale behavior while zooming.

Author: Agreed. moving in out of a section makes sense.

> With the Mouse Wheel on the Map Canvas

Reviewer: What do you think about moving some comments regarding interaction between mouse and canvas here (pan, zoom in/out)? We'd then have a mouse section instead of mouse wheel.

Author: With a table listing all actions/tools, I think we could just mention pan, zoom in/out. "In addition to the Pan, Zoom In and Zoom Out tools described above, you can use the middle mouse button to ........". We could still have a Mouse section

.. index:: Zoom, Pan, Map navigation
.. _zoom_pan:

Zooming and Panning
-------------------

There are multiple ways to zoom and pan to an area of interest.
You can use the :guilabel:`Map Navigation` toolbar, the mouse and keyboard on
the map canvas and also the menu actions from the :menuselection:`View`
menu and the :menuselection:`Layers` panel contextual menu.

.. list-table::
   :header-rows: 1
   :widths: 15 15 15 55
   :class: longtable

   * - Map Navigation Toolbar
     - View Menu
     - Layer Panel Contextual Menu
     - Usage

>    * - |pan| :sup:`Pan`

Reviewer: ```suggestion
   * - |pan| :sup:`Pan Map`
```

>      - :menuselection:`View --> Pan Map`

Author: It could be nice to show the icons in the View Menu column. I would show them smaller than in the Map Navigation Toolbar column (already the case in the View menu). Is there a way to override the image size or I need to re-declare a smaller image for each icons?

`.. |panSmall| image:: /static/common/mActionPan.png`
`:width: 1em`

>      - :menuselection:`View --> Pan Map`

Reviewer: Afaik, you'd need to declare the new references, either:
- in the substitution.rst file and add it in the substitutions list at the end of this file
- or above the substitution list, to limit their scope to this file

But I wonder if this size distinction is really necessary: up to now icons were assigned (almost) the same size, regardless they are on a toolbar or in a menu. Adding a smaller icon for menus (and applying everywhere for coherence and consistency), I'm not sure of how the doc would render but ultimately, I'm afraid it'd add burden on writers (duplicate list of icons and remember which one to use).
I suggested the separate columns for menu and toolbar because I thought some actions were exclusive to menu vs toolbar(\*) and had different label or usage. but it's not the case, so probably no need to repeat labels over the columns. What about  something like https://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/attribute_table.html#introducing-the-attribute-table-interface (_sorry to suggest another formatting. Just trying to find the best easy-to-understand-and-maintain scenario_)

\* Btw, the top message of https://github.com/qgis/QGIS/pull/40649 should be updated to reflect the actual behavior before it's harvested in 3.18 changelog

>      - :menuselection:`View --> Pan Map`

Author: Let me know what you think of the new table structure in the recent commit. I think we're getting there. I also added the "Zoom to Native Resolution" tool. However, I would appreciate a suggestion for the "Usage" column. Honestly, I have never used it and have no idea what it really does.

>      - :menuselection:`View --> Pan Map`

Reviewer: Yes LGTM. Let's get feedback for the usage and merge. Thanks @TurboGraphxBeige

>      - :menuselection:`View --> Pan Map`

Reviewer 2: "Zoom to Native Resolution" means to zoom to that zoom level, where each pixel of the active raster layer would fill one screen pixel only. This way no raster pixel get's "hidden", as it is is case when zooming to lower scales.

>      - :menuselection:`View --> Pan Map`

Reviewer 3: AFAIK: it is only for raster layers and matches 1 pixel of raster to 1 pixel of screen.

>      - :menuselection:`View --> Pan Map`

Reviewer: Thanks guys. @TurboGraphxBeige, your choice :smile:

>      - :menuselection:`View --> Pan Map`

Author: I think that's it. Feel free to rephrase it if needed.

     - :menuselection:`View --> Zoom Out`
     -
     - When activated, left click anywhere on the map canvas to zoom in or zoom
       out one level. The mouse cursor position will be the center of the
       zoomed area of interest. You can also zoom in to an area (or zoom out)
       by dragging a rectangle on the map canvas with the left mouse button.
   * - |panToSelected| :sup:`Pan Map to Selection`
     - :menuselection:`View --> Pan Map to Selection`
     -
     - Pan the map to the selected features of all the selected layers in the
       :menuselection:`Layers` panel.
   * - |zoomToSelected| :sup:`Zoom to Selection`
     - :menuselection:`View --> Zoom To Selection`
     - :menuselection:`Layers --> Zoom To Selection`
     - Zoom to the selected features of all the selected layers in the
       :menuselection:`Layers` panel.
   * - |zoomToLayer| :sup:`Zoom to Layer(s)`
     - :menuselection:`View --> Zoom To Layer(s)`
     - :menuselection:`Layers --> Zoom to Layers(s)`
     - Zoom to the extent of all the selected layers in the
       :menuselection:`Layers` panel.
   * - |zoomFull| :sup:`Zoom Full`
     - :menuselection:`View --> Zoom Full`
     -

>      - Zoom to the extent of all the visible layers in the map.

Reviewer: by visible, you mean the ones I checked the box for? Is it not the extent of all layers in the project?

Author: Oh you're right! I tested the behavior on 3.10 and I didn't know it has been changed since then.

       zoomed area of interest. You can also zoom out from an area
       by dragging a rectangle on the map canvas with the left mouse button.
     - |checkbox|
     - |checkbox|
     -
   * - |panToSelected|
     - Pan Map to Selection
     - Pan the map to the active layer's selected features.
     - |checkbox|
     - |checkbox|
     -
   * - |zoomToSelected|
     - Zoom To Selection
     - Zoom to the active layer's selected features.
     - |checkbox|
     - |checkbox|
     - |checkbox|
   * - |zoomToLayer|
     - Zoom To Layer
     - Zoom to the active layer's extent.
     - |checkbox|
     - |checkbox|
     - |checkbox|
   * - |zoomFull|
     - Zoom Full
     - Zoom to the extent of all the layers in the project.
     - |checkbox|
     - |checkbox|
     -
   * - |zoomLast|
     - Zoom Last
     - Zoom the map to the previous extent in history.
     - |checkbox|
     - |checkbox|
     -
   * - |zoomNext|
     - Zoom Next
     - Zoom the map to the next extent in history.
     - |checkbox|
     - |checkbox|
     -
   * - |zoomToNativeResolution|
     - Zoom to Native Resolution
     -
     - |checkbox|
     - |checkbox|

>      -

Reviewer: ```suggestion
     - |checkbox|
```
This is a raster feature
And I'm afraid I can't tell how it actually plays, other than it's somehow related to pixel size (if I'm not wrong).
Any raster guru around to explain this tool? Thanks.

  * :menuselection:`View --> Zoom In` will activate the |zoomIn| :sup:`Zoom In` tool available in the toolbar.

  * :menuselection:`View --> Zoom Out` will activate the |zoomOut| :sup:`Zoom Out` tool available in the toolbar.

  * :menuselection:`View --> Pan Map to Selection` will pan the map to the selected features of all the selected
    layers in the layer panel.

  * :menuselection:`View --> Zoom To Selection` will pan the map to the selected features
    of all the selected layers in the layer panel. This action is also available from the context menu when right-clicking
    on a layer in the layer panel.

  * :menuselection:`View --> Zoom To Layer(s)` will zoom to the extent of all the selected layers
    in the layer panel. This action is also available from the context menu when right-clicking on a layer in the layer panel.

  * :menuselection:`View --> Zoom Full` will zoom to the extent of all the layers of the map.

  * :menuselection:`View --> Zoom Last` will zoom the map to the previous extent in history.

  * :menuselection:`View --> Zoom Next` will zoom the map to the next extent in history.

> Note that all these actions are also available from the toolbar.

Reviewer: A good reason to merge them, isn't it? :smile:

Reviewer 4: Seems that we will need to hold this one a bit since the author says it describes stuff from 3.18?

Author: There's a bit of stuff from 3.18, indeed.

Reviewer 4: Can you split 3.18 changes into a new PR?

Author: Yes, will do.
